src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -698,11 +698,8 @@ trait TypeDiagnostics {
         } else f
       }
       def apply(tree: Tree): Tree = {
-        // Error suppression (in context.warning) would squash some of these warnings.
-        // It is presumed if you are using a -Y option you would really like to hear
-        // the warnings you've requested; thus, use reporter.warning.
         if (settings.warnDeadCode && context.unit.exists && treeOK(tree) && exprOK)
-          reporter.warning(tree.pos, "dead code following this construct")
+          context.warning(tree.pos, "dead code following this construct")
```

Comment:
I am not sure I fully understand this, but I *think* it shouldn't be a problem, based on the fact that this check is indeed run after typer, as a post-processing step, and the context reporter is the one from the root context.